arangod/Replication2/StateMachines/Document/DocumentStateTransactionHandler.cpp

```diff
@@ -26,14 +26,14 @@
 #include "Basics/application-exit.h"
 #include "Basics/voc-errors.h"
 #include "Logger/LogContextKeys.h"
-#include "Replication2/LoggerContext.h"
+#include "Replication2/StateMachines/Document/DocumentStateHandlersFactory.h"
+#include "Replication2/StateMachines/Document/DocumentStateShardHandler.h"
 #include "Replication2/StateMachines/Document/DocumentStateTransaction.h"
-#include "Transaction/ReplicatedContext.h"
-#include "DocumentStateHandlersFactory.h"
+#include "VocBase/AccessMode.h"
 
-namespace {
-
-auto shouldIgnoreError(arangodb::OperationResult const& res) noexcept -> bool {
+namespace arangodb::replication2::replicated_state::document {
+auto DocumentStateTransactionHandler::shouldIgnoreError(
+    arangodb::OperationResult const& res) noexcept -> bool {
```

Comment:
NIT: any particular reason why you moved this function into the DocumentStateTransactionHandler? I usually keep function that are not really related to a class (low cohesion, ie., do not rely on its members/functions) outside as separate functions, especially when they are not used outside of this module. This also helps keep the class' interface clean since a member function also has to be declared in the header.

Reply:
While working with concepts and visitors, I had some linker issues which I wanted to get out of the way quickly, so I moved these functions around. That's no longer the case now, hence I moved them back to an anonymous namespace in the cpp file. I agree with your points.